### PR TITLE
Add ability to override MPI variables

### PR DESCRIPTION
**Pull request by white238**
Provides a way to override default MPI behavior in a clean way.
Also converts any code snippets in the tutorial from line numbers (always breaking) to tags. #262

### CODE REVIEW

File: cmake/thirdparty/SetupMPI.cmake [hunk 1]
```diff
@@ -67,8 +69,24 @@ if (ENABLE_FIND_MPI)
     list(REMOVE_DUPLICATES _mpi_libraries)
 endif()
 
+# Allow users to override CMake's FindMPI
```
Comment (kennyweiss) [inline cmake/thirdparty/SetupMPI.cmake]: Do we always want to override, as opposed to appending?

File: docs/creating_execs_and_libs.rst [hunk 2]
```diff
@@ -26,14 +27,17 @@ like ``blank_project`` in the previous section, we can start using BLT's macros.
 Creating an executable is as simple as calling the following macro:
 
 .. literalinclude:: tutorial/calc_pi/CMakeLists.txt
+   :start-after: _blt_tutorial_example_executable_start
```
Comment (kennyweiss) [inline docs/creating_execs_and_libs.rst]: Thanks for fixing these!

File: docs/creating_execs_and_libs.rst [hunk 3]
```diff
@@ -106,3 +115,8 @@ without the linker removing unused symbols in the larger library.
     blt_add_exectuble(NAME       helloWorld
                       SOURCES    main.cpp
                       DEPENDS_ON myObjectLibrary)
+
+.. note::
+  Due to record keeping on BLT's part to make object libraries as easy to use
+  as possible, you need to define object libraries before you use them
```
Comment (kennyweiss) [inline docs/creating_execs_and_libs.rst]: Just to make sure that I understand this -- it sounds like typical CMake targets can be used before they are defined, but that is not the case for object libraries. Is that right?

File: docs/external_dependencies.rst [hunk 4]
```diff
@@ -69,33 +72,46 @@ which uses MPI to parallelize the calculation over the integration intervals.
...
 
 .. literalinclude:: tutorial/calc_pi/test_2.cpp
+   :start-after: _blt_tutorial_calcpi_test2_main_start
+   :end-before:  _blt_tutorial_calcpi_test2_main_end
    :language: cpp
-   :lines: 40-54
+
+.. note::
+  While we have tried to ensure that BLT chooses the correct setup information
+  for MPI, but there are several niche cases where the default behavior is
```
Comment (kennyweiss) [inline docs/external_dependencies.rst]: Typo -- please remove `but`

File: cmake/thirdparty/SetupMPI.cmake (hunk 1 above)
Comment (white238) [reply cmake/thirdparty/SetupMPI.cmake]: I have run into cases where the variables are wrong and cannot be solved by appending.

File: docs/creating_execs_and_libs.rst (hunk 2 above)
Comment (white238) [reply docs/creating_execs_and_libs.rst]: I actually looked for this feature when i started this and have no idea how i missed it.  So much better than line numbers.

File: docs/creating_execs_and_libs.rst (hunk 3 above)
Comment (white238) [reply docs/creating_execs_and_libs.rst]: Yes. I find it very strange but you can.

File: cmake/thirdparty/SetupMPI.cmake (hunk 1 above)
Comment (kennyweiss) [reply cmake/thirdparty/SetupMPI.cmake]: ok then.

**Pull request merged**